src/libcore/num/f64.rs

```diff
@@ -487,4 +488,135 @@ impl f64 {
+    ///
+    /// NaNs with positive sign are ordered greater than all other floating-point
+    /// values including positive infinity, while NaNs with negative sign are
+    /// ordered the least, below negative infinity. Two different NaNs with the
+    /// same sign are ordered first by whether the are signaling (signaling is
+    /// less than quiet if the sign is positive, reverse if negative) and second
+    /// by their payload interpreted as integer (reversed order if the sign is negative).
+    ///
+    /// This means all different (canonical) floating point bit patterns are
+    /// placed in a linear order, given below in ascending order:
+    ///
+    /// - Quiet Not-a-Number with negative sign (ordered by payload, descending)
+    /// - Signaling Not-a-Number with negative sign (ordered by payload, descending)
+    /// - Negative infinity
+    /// - Negative finite non-zero numbers (ordered in the usual way)
+    /// - Negative zero
+    /// - Positive zero
+    /// - Positive finite non-zero numbers (ordered in the usual way)
+    /// - Positive infinity
+    /// - Signaling Not-a-Number with positive sign (ordered by payload, ascending)
+    /// - Quiet Not-a-Number with positive sign (ordered by payload, ascending)
+    ///
+    /// # Examples
+    ///
+    /// Most follow the normal ordering:
+    /// ```
+    /// #![feature(float_total_cmp)]
+    /// use std::cmp::Ordering::*;
+    ///
+    /// assert_eq!(f64::total_cmp(1.0, 2.0), Less);
+    /// assert_eq!(f64::total_cmp(1.0, 1.0), Equal);
+    /// assert_eq!(f64::total_cmp(2.0, 1.0), Greater);
+    /// assert_eq!(f64::total_cmp(-1.0, 1.0), Less);
+    /// assert_eq!(f64::total_cmp(1.0, -1.0), Greater);
+    /// assert_eq!(f64::total_cmp(-1.0, -2.0), Greater);
+    /// assert_eq!(f64::total_cmp(-1.0, -1.0), Equal);
+    /// assert_eq!(f64::total_cmp(-2.0, -1.0), Less);
+    /// assert_eq!(f64::total_cmp(-std::f64::MAX, -1.0e9), Less);
+    /// assert_eq!(f64::total_cmp(std::f64::MAX, 1.0e9), Greater);
+    /// ```
+    ///
+    /// Zeros and NaNs:
+    /// ```
+    /// #![feature(float_total_cmp)]
+    /// use std::cmp::Ordering::*;
+    ///
+    /// assert_eq!(f64::partial_cmp(&0.0, &-0.0), Some(Equal));
+    /// assert_eq!(f64::total_cmp(-0.0, 0.0), Less);
+    /// assert_eq!(f64::total_cmp(0.0, -0.0), Greater);
+    ///
+    /// assert_eq!(f64::partial_cmp(&std::f64::NAN, &std::f64::NAN), None);
+    /// assert_eq!(f64::total_cmp(-std::f64::NAN, std::f64::NAN), Less);
+    /// assert_eq!(f64::total_cmp(std::f64::NAN, std::f64::NAN), Equal);
+    /// assert_eq!(f64::total_cmp(std::f64::NAN, -std::f64::NAN), Greater);
+    /// assert_eq!(f64::total_cmp(std::f64::NAN, std::f64::INFINITY), Greater);
+    /// assert_eq!(f64::total_cmp(-std::f64::NAN, -std::f64::INFINITY), Less);
+    /// ```
+    #[unstable(feature = "float_total_cmp", issue = "55339")]
+    #[inline]
+    pub fn total_cmp(self, other: f64) -> cmp::Ordering {
```

Comment:
Plausibly, but that's a general `f32.rs`/`f64.rs` problem not specific to this method -- the same is true of min, max, to_bits, etc -- so I will not be taking action on it in this PR.

Reply:
Just because the other methods could be improved does not mean that we have to make the situation worse with this PR. It would be better to just add a `float_total_ord.rs` module with a macro to define the method and the traits, and use that from `f32.rs`/`f64.rs`.